grid/social-media/upload-from-url/maps/facebook.suma

```diff
@@ -0,0 +1,135 @@
+profile = "social-media/upload-from-url@1.0"
+provider = "facebook"
+
+// https://developers.facebook.com/docs/video-api/guides/reels-publishing
+
+map RegisterUpload {
+  apiVersion = call GetApiVersion()
+
+  call GetPageAccessToken(pageId = input.profileId, userAccessToken = parameters.accessToken) {
+    return map error if (outcome.error) outcome.error
+    
+    token = outcome.data.pageAccessToken
+  }
+
+  http POST "video" "/{apiVersion}/{input.profileId}/videos" {
+    request {
+      body {
+        access_token = token,
+        file_url = input.url,
+        published = false,
+        description = input.caption
+      }
+    }
+
+    response 200 "application/json" {
+      map result {
+        uploadId = body.id
+      }
+    }
+  }
+}
+
+// https://developers.facebook.com/docs/graph-api/reference/video
+map GetUploadState {
+  apiVersion = call GetApiVersion()
+  
+  http GET "/{apiVersion}/{input.uploadId}" {
+    request {
+      query {
+        fields = "status",
+        access_token = parameters.accessToken
+      }
+    }
+
+    response 200 "application/json" {
+      state = (() => {
+        switch (body.status.video_status) {
+          case 'ready':
+            return 'finished';
+          case 'error':
+            return 'error';
+          case 'processing':
+            return 'inProgress'
+          default:
+            return null;
+        }
+      })();
+
+      map error if (state === null) {
```

Comment:
I prefer having an object with mapped states over IIFE, but I understand that switch/case has more flexibility. For your consideraton:

```suggestion
      states = {
        ready: 'finished',
        error: 'error',
        processing: 'inProgress',
      }
      state = states[body.status.video_status]
      
      map error if (!state) {
```

Reply:
Good point, I think the `switch` statement is a leftover from when I had more logic in there, an object/map is cleaner.

Reply:
Well, except that the linter in `station` currently doesn't include the fix and so errors out :sweat_smile: The fix is already merged in `main` AFAIK, but upgrading the linter is beyond the scope of this PR I think